js/components/portal-dashboard/question-area.tsx

```diff
@@ -34,15 +37,16 @@ export class QuestionArea extends React.PureComponent<IProps>{
             </span>
           </div>
           <div className={css.rightIcons}>
-            <a className={css.externalLinkButton} target="_blank" data-cy="open-activity-button">
+            <a className={css.externalLinkButton} href={activityURL} target="_blank" data-cy="open-activity-button">
               <LaunchIcon className={css.icon} />
             </a>
-            <div className={css.teacherEditionIcon}>
-              <a className={teacherEditionButtonClasses} target="_blank" data-cy="open-teacher-edition-button">
-                <LaunchIcon className={css.icon} />
-              </a>
-              <div className={teacherEditionBadge}>TE</div>
-            </div>
+            {hasTeacherEdition &&
+              <div className={css.teacherEditionIcon}>
+                <a className={teacherEditionButtonClasses} href={activityURL+"?mode=teacher-edition"} target="_blank" data-cy="open-teacher-edition-button">
+                  <LaunchIcon className={css.icon} />
+                </a>
+               <div className={teacherEditionBadge}>TE</div>
```

Comment:
Very minor, but check spacing.  I think the tabbing is off here and this should be one space in.

Reply:
Also, note the this div with the Teacher Edition badge is outside the clickable <a> link.  So even though the div has hover and active effects, clicking on it does nothing.  This threw me off - at first I thought the link didn't work.  We probably need to restructure this so both the icon and this div that contains the TE can be clicked.  The easiest is probably just moving this div inside of the <a> and then tweaking the CSS.

Reply:
![image](https://user-images.githubusercontent.com/5126913/93618013-942d7980-f98b-11ea-858d-3bcfee764cb3.png)

Reply:
done